sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -87,6 +87,39 @@
 MAX_BATCH_OPERATION_SIZE = 100
 
 
+def proxy_info_from_environment_var(proxy_env_var):
+  """Reads proxy info from the environment and converts to httplib2.ProxyInfo.
+  Args:
+    proxy_env_var: environment variable string to read, http_proxy or
+       https_proxy (in lower case).
+       Example: http://myproxy.domain.com:8080
+  Returns:
+    httplib2.ProxyInfo constructed from the environment string.
+  """
+  proxy_url = os.environ.get(proxy_env_var)
+  if not proxy_url or not proxy_env_var.lower().startswith('http'):
```

Comment:
<!--new_thread; commit:1033d84efb2ada3802967025d2a64d01f3649c59; resolved:0-->
Can you do the following?

```python
re.match('^https?://', proxy_env_var, flags=re.IGNORECASE)
```